fdbserver/VersionedBTree.actor.cpp

```diff
@@ -7703,10 +7714,15 @@ class KeyValueStoreRedwood : public IKeyValueStore {
 
 		EncodingType encodingType = EncodingType::XXHash64;
 
-		// Deterministically enable encryption based on uid
-		if (g_network->isSimulated() && logID.hash() % 2 == 0) {
-			encodingType = EncodingType::XOREncryption;
-			m_keyProvider = std::make_shared<XOREncryptionKeyProvider>(filename);
+		// When reopening Redwood on restart, the cluser encryption config could be unknown at this point,
+		// for which shouldEnableEncryption will return false. In that case, if the Redwood instance was encrypted
+		// before, the encoding type in the header page will be used instead.
+		//
+		// TODO(yiwu): When the cluster encryption config is available later, fail if the cluster is configured to
+		// enable encryption, but the Redwood instance is unencrypted.
+		if (encryptionKeyProvider && encryptionKeyProvider->shouldEnableEncryption()) {
+			encodingType = EncodingType::AESEncryptionV1;
```

Comment:
For better understanding: the comments "the encoding type in the header page will be used instead", however, we are hardcoding the value to AESEncryptionV1 instead of reading the header?

The usecase that comes to mind is, an attacker might attempt to replace disk with some unknown content and header value maybe unsupported to start with?

Reply:
> This line is setting the desired encoding type before opening the DWALPager. The DWALPager will read the header page to update the encoding type if needed.

If value is getting updated at later point, any reason we want to override the value? Is it possible that override doesn't happen and encryption encoding is set to a wrong value?